vscode extension updates (#1365)
demisto/content-docs: docs/concepts/vscode-extension.md

### Local development (Linux, MacOS, WSL2)

> The VScode extension supports fully setting up your development environment automatically.

Reviewer: ```suggestion
The VScode extension supports setting up your development environment automatically.
```

#### Usage

Execute the command **_XSOAR: install local development environment_**, either from [VSCode Command Pallete](https://code.visualstudio.com/docs/getstarted/userinterface#_command-palette), or by right-clicking a file.

If you want to install the dependencies manually, follow the instructions in this [guide](./dev-setup.md#option-3-manual-setup) until the `Bootstrap` step.

## Setup integrations and scripts environment

> Each integration or script in `XSOAR` runs on a different environnement, and has different dependencies.

Reviewer: ```suggestion
Each integration or script in `Cortex XSOAR` runs on a different environment, and has different dependencies.
```

### Usage

- Go to the integration or a script.
- Right-click it, and select **Setup integration/script environment**
- There will be a popup asking you if you want to use the current workspace or open a new one with a virtual environment.
  - Using the current workspace is quicker, but you may will not have autocompletion for some integrations.
  - Opening a new workspace will take longer, but you will have autocompletion for all integrations.
- With both options, you will be able to debug your integration/script easily, run unit tests, and see problems in the IDE.

## Debugging

- First, [Setup integrations and scripts environment](#setup-integrations-and-scripts-environment)
- Then, read the [Debugging using your IDE](../integrations/debugging#Debugging%20using%20your%20IDE) section.
- Go to the Run and Debug (⇧⌘D), and make sure that `Docker: debug (<integration>)` is selected.
- Click on the green arrow or F5 to start debugging.

## Run and Debug tests

- First, [Setup integrations and scripts environment](#setup-integrations-and-scripts-environment)

> - Go to the Run and Debug (⇧⌘D), and make sure that `Docker: debug tests (<integration>)` is selected.

Reviewer: What is this?  (⇧⌘D)

Author: This is a shorcut for it (Command+Shift+D)
I will add a link to the offcial docs for it:
https://code.visualstudio.com/docs/editor/debugging#_run-and-debug-view

- Click on the green arrow or F5 to start debugging.

## Notes

> - If during the installation one or more Python packages fail to install, the installation will proceed and create the virtual environment with the packages that were installed.

Reviewer: ```suggestion
If during the installation one or more Python packages fail to install, the installation will proceed and create the virtual environment with the packages that were installed.
```

## Troubleshooting

### **Setup integration/script environment** fails

* Make sure **Docker** is running.
* Make sure that `Allow the default Docker socket to be used (requires password)` is enabled in **Docker** advanced settings.

> * If _Docker_ is running try to [clean up the Docker](https://docs.docker.com/config/pruning/) or [sign in to docker](https://www.docker.com/blog/seamless-sign-in-with-docker-desktop-4-4-2/) to avoid the [docker pull rate limit](https://docs.docker.com/docker-hub/download-rate-limit/#:~:text=Pull%20rates%20limits%20are%20based,to%205000%20pulls%20per%20day.).

Reviewer: ```suggestion
* If _Docker_ is running try to [clean up Docker](https://docs.docker.com/config/pruning/) or [sign in to Docker](https://www.docker.com/blog/seamless-sign-in-with-docker-desktop-4-4-2/) to avoid the [Docker pull rate limit](https://docs.docker.com/docker-hub/download-rate-limit/#:~:text=Pull%20rates%20limits%20are%20based,to%205000%20pulls%20per%20day.).
```
